src/docs/options.md

```diff
@@ -117,6 +117,38 @@ The options available in a given Pants-enabled codebase depend on which backends
 + To see global (or goal-specific) basic options, enter `./pants help (goal)`.
 + To see global (or goal-specific) basic and advanced options, enter `./pants help-advanced (goal)`.
 
+### Compiler Options
+
+Compiler options are set the same way any options are set. Please remember to prefix the flag so
+that it is passed through to the target compiler. For instance, Zinc requires the prefix "-S" to
+pass a flag to scalac.
+
+A common example is enabling statistics in scalac (through Zinc), but Zinc doesn't provide the
+output. In order to enable and view scalac's statistics with Pants, add the statistics flag to
+`compiler_option_sets_enabled_args` in the `pants.ini` file (see below) like so:
+
+    :::ini
+    compiler_option_sets_enabled_args: {
+        'c_profile': [ '-S-Ystatistics' ],
+      }
+
+Then, refer to the set in the target that needs analyzing like below:
+
+    :::python
+    scala_library(
+      sources = globs("*.scala"),
+      compiler_option_sets = { "c_profile" },
+      dependencies = [
+        ...
+      ],
+    )
+
+Now the target is ready to compile, but the output will be squelched by Zinc. To avoid this, add
```

Comment:
This might not be a good example to use... maybe demonstrate with some other flag in order to avoid needing to skip this extra step?

Reply:
I used this example because Shane spent a lot of time (like 2 days) trying to figure out why his output wasn't showing, and I was trying to debug something else he was working on, so I thought it would be good to reference this somewhere in the docs so that the next time something like this happens, it is documented.